overlord/fdestate/fdemgr.go

```diff
@@ -181,6 +188,85 @@ func (m *unlockedStateManager) Unlock() (relock func()) {
 	return func() {}
 }
 
+// EncryptedDisk implements backend.EncryptedDisk
+type EncryptedDisk struct {
+	// UUID is the uuid of the luks2 header of the disk
+	UUID string
+	// Role is container role of the disk. See KeyslotRoleInfo.Parameters.
+	Role string
+	// LegacyKeys gives path the legacy keys index by the key name that matches the token key
+	LegacyKeys map[string]string
+}
+
+func (disk *EncryptedDisk) GetRole() string {
+	return disk.Role
+}
+
+func (disk *EncryptedDisk) GetLegacyKeys() map[string]string {
+	return disk.LegacyKeys
+}
+
+func (disk *EncryptedDisk) GetDevPath() string {
+	return fmt.Sprintf("/dev/disk/by-uuid/%s", disk.UUID)
+}
+
+// GetEncryptedDisks returns the disk
+func (m *FDEManager) GetEncryptedDisks() ([]backend.EncryptedDisk, error) {
+	return getEncryptedDisks(m.state)
+}
+
+func getEncryptedDisks(state *state.State) ([]backend.EncryptedDisk, error) {
```

Comment:
it's the order first data then save important? it's seems maybe the case looking at the code in fdestate but not sure, either way this probably needs a comment about what it returns either way

Reply:
Added it on the doc for the public function.